crates/parcel_plugin_transformer_js/src/transformer.rs

```diff
@@ -1,24 +1,894 @@
+  transformer_config: &parcel_js_swc_core::Config,
+  result: parcel_js_swc_core::TransformResult,
+  options: &ParcelOptions,
+) -> Result<TransformResult, Vec<Diagnostic>> {
+  let asset_file_path = asset.file_path.to_path_buf();
+  let asset_environment = asset.env.clone();
+  let asset_id = asset.id();
+
+  if let Some(shebang) = result.shebang {
+    asset.set_interpreter(shebang);
+  }
+
+  let (mut dependency_by_specifier, invalidate_on_file_change) = convert_dependencies(
+    transformer_config,
+    result.dependencies,
+    &asset_file_path,
+    &asset_environment,
+    asset_id,
+  )?;
+
+  if result.needs_esm_helpers {
+    let mut dependency_flags = DependencyFlags::empty();
+    dependency_flags.set(
+      DependencyFlags::HAS_SYMBOLS,
+      result.hoist_result.is_some() || result.symbol_result.is_some(),
+    );
+    let dependency = make_esm_helpers_dependency(
+      options,
+      &asset_file_path,
+      asset_environment,
+      dependency_flags,
+      asset_id,
+    );
+    dependency_by_specifier.insert(dependency.specifier.as_str().into(), dependency);
+  }
+
+  if let Some(hoist_result) = result.hoist_result {
+    asset.flags.set(AssetFlags::HAS_SYMBOLS, true);
+    // Pre-allocate expected symbols
+    asset
+      .symbols
+      .reserve(hoist_result.exported_symbols.len() + hoist_result.re_exports.len() + 1);
+
+    // Collect all exported variable names into `asset.symbols`
+    for symbol in &hoist_result.exported_symbols {
+      let symbol = transformer_exported_symbol_into_symbol(&asset_file_path, &symbol);
+      asset.symbols.push(symbol);
+    }
+
+    // Collect all imported symbols into each of the corresponding dependencies' symbols array
+    for symbol in hoist_result.imported_symbols {
+      if let Some(dependency) = dependency_by_specifier.get_mut(&symbol.source) {
+        let symbol = transformer_imported_symbol_to_symbol(&asset_file_path, &symbol);
+        dependency.symbols.push(symbol);
+      }
+    }
+
+    for symbol in hoist_result.re_exports {
+      if let Some(dependency) = dependency_by_specifier.get_mut(&symbol.source) {
+        if is_re_export_all_symbol(&symbol) {
+          let loc = Some(convert_loc(asset_file_path.clone(), &symbol.loc));
+          dependency.symbols.push(make_export_all_symbol(loc));
+          // TODO: Why isn't this added to the asset.symbols array?
+        } else {
+          let existing = dependency
+            .symbols
+            .as_slice()
+            .iter()
+            .find(|candidate| candidate.exported == &*symbol.imported);
+          let existing_flags = existing
+            .map(|e| e.flags & SymbolFlags::IS_WEAK)
+            .unwrap_or(SymbolFlags::IS_WEAK);
+
+          // TODO: What is this? There is no such mangling in the transformer
+          let re_export_name = existing
+            .map(|sym| sym.local.clone())
+            .unwrap_or_else(|| format!("${:016x}$re_export${}", asset_id, symbol.local).into());
```

Comment:
What happened here? Because this symbol is not in the transformer.

Reply:
This represents an `export {foo} from '...'` statement. In that case there is no "local" symbol used in the source code, the asset only passes through the exports and does not use them itself. So the synthetic local name here is only so the asset symbol and dependency symbol can be matched up.

Another case of a non-ideal symbol representation that we should change eventually but we need for compatibility with JS packager for now.

Reply:
probably could be, I just ported the JS version https://github.com/parcel-bundler/parcel/blob/c03a29ede215967a9ec4e472a1161eed0a913af6/packages/transformers/js/src/JSTransformer.js#L907-L911